packages/monitoring-system/data-processor/test/unit/data-processors/cloud-logs-data-processor.spec.ts

```diff
@@ -0,0 +1,601 @@
+    const jsonLines = lines.map(line => JSON.parse(line));
+    assert(jsonLines.length > 0, 'No logs found');
+    let foundErrorLog = false;
+    for (const line of jsonLines) {
+      const isErrorLevel = line.level === 50;
+      const hasCorrectMsg =
+        line.message && String(line.message).includes(expectedErrorMsg);
+      foundErrorLog = isErrorLevel && hasCorrectMsg;
+      if (foundErrorLog) break;
+    }
+    assert(foundErrorLog, 'No relevant errors were logged');
+  } catch (error) {
+    throw new Error(`Failed to read stream: ${error}`);
+  }
+}
+
+/**
+ * Asserts that the given message was properly processed and acked.
+ * @param message message to test
+ * @param expectedRecords expected records in firestore
+ * @param preExistingRecords records to add to firestore before test
+ */
+async function testValidMessage(
+  message: {},
+  expectedRecords: MockRecord[],
+  preExistingRecords?: MockRecord[]
+): Promise<void> {
+  if (preExistingRecords) {
+    preExistingRecords.forEach(record => mockFirestore.addRecord(record));
+  }
+  return startAndSendMessage(message).then(messageId => {
+    assert(mockSubscription.wasAcked(messageId));
+    expectedRecords.forEach(record => mockFirestore.assertRecord(record));
+  });
+}
+
+/**
+ * Asserts that the given malformed message was properly processed
+ * @param malformedMessage a malformed message
+ * @param writeStream the stream to which logs are written
+ */
+async function testMalformedMessage(
+  malformedMessage: {},
+  expectedErrorMsg: string,
+  writeStream: ObjectWritableMock
+): Promise<void> {
+  return startAndSendMessage(malformedMessage).then(messageId => {
+    assert(mockSubscription.wasAcked(messageId));
+    assertErrorLogged(expectedErrorMsg, writeStream);
+  });
+}
+
+/**
+ * Starts the processor and sends a message
+ * @param message message to send
+ * @returns promise with message id
+ */
+async function startAndSendMessage(message: {}): Promise<string> {
+  return startAndSendMultipleMessages([message]).then(ids => ids[0]);
+}
+
+/**
+ * Starts the processor and sends multiple messages
+ * @param messages messages to send
+ * @returns promise with an array of message ids
+ */
+async function startAndSendMultipleMessages(
+  messages: Array<{}>
+): Promise<string[]> {
+  const processingTask = processor.collectAndProcess();
+  const allMessages: Array<Promise<string>> = messages.map(message => {
+    const messageId = mockSubscription.sendMockMessage(asBuffer(message));
+    return processingTask.then(() => {
+      return messageId;
+    });
+  });
+  return Promise.all(allMessages);
+}
+
+describe('Cloud Logs Processor', () => {
+  describe('collectAndProcess()', () => {
+    beforeEach(() => {
+      mockFirestore = new MockFirestore({
+        Bot_Execution: {},
+        Error: {},
+        Trigger: {},
+        Action: {},
+        Action_Type: {},
+        GitHub_Repository: {},
+        GitHub_Object: {},
```

Comment:
Are you asking if the default should be empty collections or if this should be the default data in `MockFirestore` class?

Reply:
The latter, should MockFirestore handle this for you?

Reply:
`MockFirestore` is designed as a general purpose mock instead of a mock for the Data Processor's Firestore. So I didn't want to add Data Processor specific things into `MockFirestore` itself because I wanted to keep it as reusable as possible.

That being said, I could also create a `MockDataProcessorFirestore` that `extends MockFirestore` to add this default data, but it might just add more layers and complexity.